src/WebJobs.Script.WebHost/BlobStorageScaleMetricsRepository.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.Script.Scale;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public class BlobStorageScaleMetricsRepository : IScaleMetricsRepository
+    {
+        private readonly IConfiguration _configuration;
+        private readonly IHostIdProvider _hostIdProvider;
+        private readonly ScaleOptions _scaleOptions;
+        private readonly ILogger _logger;
+        private CloudBlockBlob _metricsBlob;
+
+        public BlobStorageScaleMetricsRepository(IConfiguration configuration, IHostIdProvider hostIdProvider, IOptions<ScaleOptions> scaleOptions, ILoggerFactory loggerFactory)
+        {
+            _configuration = configuration;
+            _hostIdProvider = hostIdProvider;
+            _scaleOptions = scaleOptions.Value;
+            _logger = loggerFactory.CreateLogger(ScriptConstants.TraceSourceScale);
```

Comment:
Rather than use custom categories, we're making all new logs use `ILogger<T>` so it's more immediately clear where these come from.

Reply:
This does have the benefit of allowing us to filter to see just scale logs, which is nice. That capability seems preferable to me than seeing these internal class names as a source.